src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -177,67 +172,34 @@ protected IRequestExecutionQueue<TRequestContext> GetRequestExecutionQueue()
 
-        public DelegatingEntryPoint(string method, AbstractTypeRefResolver typeRefResolver, IGrouping<string, RequestHandlerMetadata> handlersForMethod)
+        public DelegatingEntryPoint(string method)
         {
             _method = method;
-            _typeRefResolver = typeRefResolver;
-            _languageEntryPoint = new Lazy<FrozenDictionary<string, (MethodInfo, RequestHandlerMetadata)>>(() =>
-            {
-                var handlerEntryPoints = new Dictionary<string, (MethodInfo, RequestHandlerMetadata)>();
-                foreach (var metadata in handlersForMethod)
-                {
-                    var noValueType = NoValue.Instance.GetType();
-
-                    var requestType = metadata.RequestTypeRef is TypeRef requestTypeRef
-                        ? _typeRefResolver.Resolve(requestTypeRef) ?? noValueType
-                        : noValueType;
-                    var responseType = metadata.ResponseTypeRef is TypeRef responseTypeRef
-                        ? _typeRefResolver.Resolve(responseTypeRef) ?? noValueType
-                        : noValueType;
-                    var methodInfo = s_queueExecuteAsyncMethod.MakeGenericMethod(requestType, responseType);
-                    handlerEntryPoints[metadata.Language] = (methodInfo, metadata);
-                }
-
-                return handlerEntryPoints.ToFrozenDictionary();
-            });
         }
 
         public abstract MethodInfo GetEntryPoint(bool hasParameter);
 
-        protected (MethodInfo MethodInfo, RequestHandlerMetadata Metadata) GetMethodInfo(string language)
-        {
-            if (!_languageEntryPoint.Value.TryGetValue(language, out var requestInfo)
-                && !_languageEntryPoint.Value.TryGetValue(LanguageServerConstants.DefaultLanguageName, out requestInfo))
-            {
-                throw new InvalidOperationException($"No default or language specific handler was found for {_method} and document with language {language}");
-            }
-
-            return requestInfo;
-        }
-
         protected async Task<object?> InvokeAsync(
-            MethodInfo methodInfo,
             IRequestExecutionQueue<TRequestContext> queue,
             object? requestObject,
-            string language,
             ILspServices lspServices,
             CancellationToken cancellationToken)
         {
-            var task = methodInfo.Invoke(queue, [requestObject, _method, language, lspServices, cancellationToken]) as Task
-                ?? throw new InvalidOperationException($"Queue result task cannot be null");
-            await task.ConfigureAwait(false);
-            var resultProperty = task.GetType().GetProperty("Result") ?? throw new InvalidOperationException("Result property on task cannot be null");
-            var result = resultProperty.GetValue(task);
+            var result = await queue.ExecuteAsync(requestObject, _method, lspServices, cancellationToken).ConfigureAwait(false);
```

Comment:
Deserialization and things moved into the queue, had to move the reflection invocation down into the queue as well